python/lsst/validate/base/datum.py

```diff
@@ -131,8 +132,7 @@ def __init__(self, quantity=None, unit=None, label=None, description=None):
         if quantity is None:
             self._quantity = None
         elif isinstance(quantity, u.Quantity) or \
-                isinstance(quantity, basestring) or \
-                isinstance(quantity, bool) or isinstance(quantity, int):
+                QuantityAttributeMixin._is_non_quantity_type(quantity):
```

Comment:
Yeah, the first`if` saves it I think. Slightly dangerous but I'm confident in the level of unit testing for this.